test/sanitize_json.d

```diff
@@ -0,0 +1,231 @@
+import std.typecons : Flag, Yes, No;
+import std.array : Appender, appender;
+import std.bigint;
+import std.internal.cstring : tempCString;
+import std.string : startsWith, replace;
+import std.format : formattedWrite, format;
+import std.file : readText;
+import std.stdio;
+
+import json;
+
+__gshared File outFile;
+
+void usage()
+{
+    writeln("Usage: santize_json <input-json> [<output-json>]");
+}
+int main(string[] args)
+{
+    args = args[1 .. $];
+    if (args.length == 0)
+    {
+        usage();
+        return 1;
+    }
+    string inFilename = args[0];
+    if(args.length == 1)
+    {
+        outFile = stdout;
+    }
+    else if(args.length == 2)
+    {
+        outFile = File(args[1], "w");
+    }
+    else
+    {
+        writeln("Error: too many command line arguments");
+        return 1;
+    }
+
+    string jsonText;
+    {
+        auto inFile = File(inFilename, "r");
+        auto inFileSizeLong = inFile.size;
+        assert(inFileSizeLong < size_t.max, "file too large");
+        size_t inFileSize = cast(size_t)inFileSizeLong;
+        char[] buffer = new char[inFileSize + 1];
+        assert(inFileSize == inFile.rawRead(buffer[0..inFileSize]).length);
+        buffer[inFileSize] = '\0';
```

Comment:
Why is this necessary. You don't use any C function?

Reply:
The JSON parser relies on the file content being null-terminated.  Makes the parser simpler.